src/main/java/com/slisenko/education/html/DownloadException.java

```diff
@@ -0,0 +1,12 @@
+package com.slisenko.education.html;
+
+/**
+ * Exception class used in case of errors when downloading a page to a file
+ */
+public class DownloadException extends Exception {
+
+    public DownloadException(String message) {
```

Comment:
please also pass parent exception as a constructor parameter and wrap it
use the constructor of the Exception class wich accepts Exception or Throwable

Reply:
Correct remark, the root exception is not lost.
Fixed.